### ISSUE TITLE
Tests on Logistic Regression

### ISSUE THREAD

**Issue by JeremyRieussec** [JeremyRieussec/TrustRegionTests.jl#12]
- [ ] Generate logit population 

Implement code for : 
- [ ] function 
- [ ] gradient and gradients  
- [ ] Hessian 
- [ ] Hessian-vector product

Test with full gradient and Hessian
- [ ] Gradient Descent 
- [ ] Momentum 
- [ ] Nesterov
- [ ] Adam
- [ ] Trust-region with various updates
